### PR TITLE
Add WeightNorm wrapper layer

### PR DESCRIPTION
**Pull request by seanpmorgan**
Adds a wrapper for Weight Normalization as requested in #14070 & #10125 .
Contains optional data dependent initialization for eager execution, and works on both keras.layers and tf.layers

I struggled to figure out where to place this Wrapper, as no other layers in contrib appear to subclass anything from tf.Keras, but going forward I believe this is the direction TF is headed.  Please advise if it should go somewhere else (maybe a wrappers module in contrib?)

Collab Example:
https://colab.research.google.com/drive/1nBQSAA78oUBmi9fhnHJ_zWhHq2NXjwIc#scrollTo=au25bSP75hdr

Wrapped graph:
selection_002

### CODE REVIEW

**Review by ctrlaltvikas — COMMENTED** [review]
haha. looks good.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by dalexander** [comment]
+1

Can this be reviewed/merged?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by ctrlaltvikas — APPROVED** [review]

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 105 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by alexbeloi** [comment]
+1

I'm already using this, it would be nice to see it in an official release. Thanks for all the work from contributors and reviewers.

**Comment by seanpmorgan** [comment]
This will be moved to a PR for [tensorflow/addons](https://github.com/tensorflow/addons).

Closing for now to stop the nagging, but I'll reference this issue within a couple of weeks.

**Comment by kaczmarj** [comment]
a weight normalization wrapper is available in https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/wrappers.py
